extensions/pyRevitTools.extension/pyRevit.tab/Drawing Set.panel/views.stack/Legends.pulldown/Copy Legends to Other Documents.pushbutton/script.py

```diff
@@ -96,3 +114,44 @@ def OnDuplicateTypeNamesFound(self, args):
                         'Legend already exists. Renaming to: "%s"', new_name)
                 revit.update.set_name(dest_view, new_name)
                 dest_view.Scale = src_legend.Scale
+
+                # matching reference planes
+                for src_plane in reference_planes:
+                    src_param = \
+                        src_plane.GetParameter(
+                            DB.ParameterTypeId.ClineSubcategory)
+                    src_subcat = \
+                        DB.Category.GetCategory(
+                            revit.doc, src_param.AsElementId())
+                    dest_plane = \
+                        dest_doc.Create.NewReferencePlane(
+                            src_plane.BubbleEnd, 
+                            src_plane.FreeEnd, 
+                            DB.XYZ(0,0,1), 
+                            dest_view)
+                    if src_subcat.Id != src_plane_category.Id:
+                        if src_subcat.Name not in dest_subcats:
+                            p_line_style = DB.GraphicsStyleType.Projection
+                            dest_subcats[src_subcat.Name] = \
```

Comment:
Instead of creating the objects directly inside the dictionaries, use variables to hold them and add them to the dictionaries only at the end.
If you combine this with the comment above, you can create a function that returns the two objects, and the if branch only adds them to the dictionaries:

```
if src_subcat.Name nor in dest_subcats:
    subcat, pat = create_dest_subcat(....)
    dest_subcats[src_subcat.Name] = subcat
    dest_pats[src_pat.Name] = pat
```

Reply:
I reworked this bit of code a bit differently than you suggested. The weird thing about this construct is that if the function returns a solid line pattern, it would get added to the dictionary, which is not desirable. Committing the changes soon...